tests/bsim/sh_common.source

```diff
@@ -12,5 +18,31 @@ function wait_for_background_jobs(){
   for process_id in $_process_ids; do
     wait $process_id || let "exit_code=$?"
   done
-  exit $exit_code #the last exit code != 0
+  [ $exit_code -eq 0 ] || exit $exit_code
+}
+
+trap ctrl_c INT
+
+function ctrl_c() {
+  echo "Aborted by CTRL-C"
+
+  for process_id in $_process_ids; do
+    kill -15 $process_id
+  done
+}
+
+function check_program_exists() {
+  if [ ! -f $1 ]; then
+    echo -e "  \e[91m`pwd`/`basename $1` cannot be found (did you forget to\
+ compile it?)\e[39m"
+    exit 1
+  fi
 }
+
+function Execute() {
+  EXECUTE_TIMEOUT="${EXECUTE_TIMEOUT:-30}"
+
+  check_program_exists $1
+  run_in_background timeout ${EXECUTE_TIMEOUT} $@
+}
```

Comment:
Since this is now being moved to a generic place, could I perhaps request that we also add a valgrind argument/environment variable? 

The reason for this is that you (or at least I) want to run Valgrind on tests for debugging purposes, but not for all tests. Usually I'd just add it in the specific shell script that I am debugging, but now it seems to only be modified in a generic way. 

So I'd suggest a way to set e.g. VALGRIND=1 and add `valgrind` to this line like
```bash
run_in_background timeout ${EXECUTE_TIMEOUT} valgrind $@
```

if it is set

Reply:
@Thalley by the way, you can just add the valgrind directly after the Execute in the script itself (that may be faster and easier than setting a variable if you just want to run one test in valgrind).

Reply:
> @Thalley by the way, you can just add the valgrind directly after the Execute in the script itself (that may be faster and easier than setting a variable if you just want to run one test in valgrind).

That is a good point, I should be able to just do `Execute valgrind ...`